contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -123,26 +132,40 @@ def execute(self):
 
       # Construct the mypy command line.
       cmd = ['--python-version={}'.format(interpreter_for_targets.identity.python)]
+
       if self.get_options().config_file:
-        cmd.append('--config-file={}'.format(os.path.join(get_buildroot(),
-                                                          self.get_options().config_file)))
+        config_file_path = os.path.join(get_buildroot(), self.get_options().config_file)
+        cmd.append('--config-file={}'.format(config_file_path))
+
       cmd.extend(self.get_passthru_args())
       cmd.append('@{}'.format(sources_list_path))
       self.context.log.debug('mypy command: {}'.format(' '.join(cmd)))
 
       # Collect source roots for the targets being checked.
-      source_roots = self._collect_source_roots()
+      env = {}
+      if self.get_options().search_source_roots:
+        source_roots = self._collect_source_roots()
+        mypy_path = os.pathsep.join([os.path.join(get_buildroot(), root) for root in source_roots])
+        env['MYPYPATH'] = mypy_path
+        self.context.log.debug('setting MYPYPATH to {}'.format(mypy_path))
 
-      mypy_path = os.pathsep.join([os.path.join(get_buildroot(), root) for root in source_roots])
-
-      # Execute mypy.
       with self.context.new_workunit(
         name='check',
         labels=[WorkUnitLabel.TOOL, WorkUnitLabel.RUN],
         log_config=WorkUnit.LogConfig(level=self.get_options().level,
                                       colors=self.get_options().colors),
         cmd=' '.join(cmd)) as workunit:
         returncode = self._run_mypy(py3_interpreter, cmd,
-          env={'MYPYPATH': mypy_path}, stdout=workunit.output('stdout'), stderr=subprocess.STDOUT)
+                                    env=env, stdout=workunit.output('stdout'), stderr=subprocess.STDOUT)
         if returncode != 0:
           raise TaskError('mypy failed: code={}'.format(returncode))
+
+  def execute(self):
+    if self.get_options().skip:
+      self.context.log.debug('mypy disabled')
+      return
+
+    python_targets = self.get_targets(self.is_non_synthetic_python_target)
```

Comment:
The `self.is_non_synthetic_python_target` check gets recomputed in `calculate_sources`... can probably drop it there.